torch/csrc/autograd/variable.cpp

```diff
@@ -25,6 +26,19 @@
 namespace torch {
 namespace autograd {
 
+// Returns a ViewFunc with a corresponding view that matches the shape,
+// stride, and storage offset of the given tensor. Requires the generated
+// AsStridedViewFunc to be available, which may not be the case (e.g. for
+// mobile).
+static std::shared_ptr<ViewFunc> create_view_func_matching(const Variable& t) {
+#ifdef AS_STRIDED_VIEW_FUNC_AVAILABLE
```

Comment:
this addresses a mobile build failure. My understanding is that mobile operates with a limited set of ops, and `as_strided()` is not one of them, so the codegen for `AsStridedViewFunc` doesn't happen.

Reply:
Did you add this flag or was it already there?
tbh I would rather have a mobile specific guard here to preserve the current (broken) state of: mobile doesn't build most ops but does build half of the autograd source code and so it is expected that we have weird slicing of the autograd codebase for mobile (because half of autograd is not available to the other half on mobile).

Reply:
yep I added this flag, but I can certainly remove it and do a mobile-specific guard here.